sdk/storage/microsoft-azure-storage-blob/src/main/java/com/microsoft/azure/storage/blob/PageBlobURL.java

```diff
@@ -516,9 +517,7 @@ public PageBlobURL withSnapshot(String snapshot) throws MalformedURLException, U
         accessConditions = accessConditions == null ? new BlobAccessConditions() : accessConditions;
         context = context == null ? Context.NONE : context;
 
-        if (prevSnapshot == null) {
-            throw new IllegalArgumentException("prevSnapshot cannot be null");
-        }
+        Objects.requireNonNull(prevSnapshot, "'prevSnapshot' cannot be null.")
```

Comment:
The packages that start with `microsoft-azure-*` are owned by the service team rather than us. We own the ones that are start with `azure-*`. Consequently, we try not to change existing behaviour of their client libraries. It may throw off customers if they were expecting an IllegalArgumentException but now it throws a NullPointerException.

Could you please revert this?

I just noticed the difference in packages, apologies.